sound/soc/sof/trace.c

```diff
@@ -154,15 +163,24 @@ int snd_sof_init_trace_ipc(struct snd_sof_dev *sdev)
 	if (sdev->dtrace_is_enabled)
 		return -EINVAL;
 
+	/* update init_offset, and reinitial offset to 0 */
+	sdev->trace_init_offset = (sdev->trace_init_offset +
+				   sdev->trace_offset) %
+				   sdev->dmatb.bytes;
+	sdev->trace_offset = 0;
+
 	/* set IPC parameters */
 	params.hdr.size = sizeof(params);
 	params.hdr.cmd = SOF_IPC_GLB_TRACE_MSG | SOF_IPC_TRACE_DMA_PARAMS;
 	params.buffer.phy_addr = sdev->dmatp.addr;
 	params.buffer.size = sdev->dmatb.bytes;
-	params.buffer.offset = 0;
 	params.buffer.pages = sdev->dma_trace_pages;
 
-	sdev->host_offset = 0;
+	/*
+	 * send offset to FW to require copying new logs
+	 * start from new offset if possible.
+	 */
+	params.buffer.offset = sdev->trace_init_offset;
```

Comment:
I am now officially confused. I understand trace_init_offset and the offset to be used on resume. Now you are using it as a dynamic value that's passed to firmware, not sure I get what you are trying to do.

Reply:
This is used to cover that case on SKL- platforms where we don't have host DMA, as I mentioned in the PR commit message:

Todo: for SKL- platforms where not using BDL entries, FW to support
non-0 offset need to be implemented for those platforms, please refer
to this:
thesofproject/sof#544

Reply:
Then let's be clear: **I will not merge a hardware-agnostic kernel patch that will only work with SKL+ firmware**. Our team has the charter to deal with legacy platforms so the firmware changes need to happen first.

Reply:
hmm, this was the biggest concern from me when writing this patch, let me raise #544 and try if we can speed up the implementation in FW side.

Reply:
Something is wrong with your references, torvalds#544 does not refer to any firmware issues